### PR TITLE
Adding Template : ansible-tower-rhel

### PR DESCRIPTION
**Pull request by amalik99**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@amalik99__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by amalik99** [comment]
Travis CI Validation Completed Sucessfully, 2 consideration with respect to this.
 - Final value at Line 761   would be  "[concat(parameters('_artifactslocation'),'scripts/', variables('scriptFileName'),parameters('_artifactsLocationSasToken'))]" , using raw script file location for now as travis wasnt translating it to user branch url on runtime.

 - artifactsLocationSasToken value is blank which is as expected, Travis CI validation highlighting it as there's no value for this parameter. Add blank space as value for completing travis validation.

**Comment by amalik99** [comment]
Template updated with best practices, Default value of _artifactslocation parameter value needs to be updated to "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/ansible-tower-rhel/" . Kept it to branch URI for now as travis wasnt translating it to user branch url on runtime.

**Comment by amalik99** [comment]
Template updated with recommendations. Please review

File: ansible-tower-rhel/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+Once deployment finishes, you will able to directly access fully configured tower GUI and connect with Ansible Clients. 
+
+##Deployment Solution Architecture 
+
+This template will deploy: 
+
+-   3 storage accounts
+-	One Virtual Network with two subnets
+-	Two Network Security Group, one for each subnet
```
Comment (bmoore-msft) [inline ansible-tower-rhel/README.md]: s/b "Groups"

File: ansible-tower-rhel/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+![Deployment Solution Architecture](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/ansible-tower-rhel/images/barracuda-architecture.png?raw=true)
+
+##Licenses and Costs 
+
+This uses RHEL 7.2 image which is a PAY AS YOU GO image and doesn't require the user to license it, it will be licensed automatically after the instance is launched first time and user will be charged hourly in addition to Microsoft's Linux VM rates.  Click [here](https://azure.microsoft.com/en-gb/pricing/details/virtual-machines/linux/#red-hat) for pricing details.
+
+##Prerequisites 
+
+- Azure Subscription with specified payment method (RHEL 7.2 is a market place product and requires payment method to be specified in Azure Subscription
+- User account with admin or contributor access to the given azure subscription
```
Comment (bmoore-msft) [inline ansible-tower-rhel/README.md]: is this needed for anything other than deployment?

File: ansible-tower-rhel/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+- Ansible Tower License( You can get a free trial license from [here](https://www.ansible.com/license) )
+
+##Deployment Steps  
+
+Build your Ansible Tower environment on Azure in a few simple steps: 
+
+- Launch the Template by clicking on Deploy to Azure.  
+- Fill in all the required parameter values. Accept the terms and condition on click Purchase. The deployment takes about 20 minutes.
```
Comment (bmoore-msft) [inline ansible-tower-rhel/README.md]: "and click Purchase"

File: ansible-tower-rhel/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+- User account with admin or contributor access to the given azure subscription
+- Ansible Tower License( You can get a free trial license from [here](https://www.ansible.com/license) )
+
+##Deployment Steps  
+
+Build your Ansible Tower environment on Azure in a few simple steps: 
+
+- Launch the Template by clicking on Deploy to Azure.  
+- Fill in all the required parameter values. Accept the terms and condition on click Purchase. The deployment takes about 20 minutes. 
+- Access the deployment job once it is successful. In deployment job output you will find the Public IP Address of client and tower VMs so you can easily connect to the VMs. Make a note of the public of Tower VM.
```
Comment (bmoore-msft) [inline ansible-tower-rhel/README.md]: it's easier to get these from the RG blade...  perhaps add an "or" option for retrieving them

File: ansible-tower-rhel/README.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+- Access Ansible GUI with the public ip of tower vm noted in above step by accessing http://publicip
+- Login with username as ‘admin’ and password specified in parameters during deployment.
```
Comment (bmoore-msft) [inline ansible-tower-rhel/README.md]: can I change the username during deployment?

File: ansible-tower-rhel/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,728 @@
...
+      "_artifactsLocationSasToken":{
+         "type":"securestring",
+         "metadata":{
+            "description":"The sasToken required to access _artifactsLocation.  When the template is deployed using the accompanying scripts, a sasToken will be automatically generated.Leave blank if unsure"
+         },
+         "defaultValue":""
+      }
+   },
+   "variables":{
+      "ansibleTowerVmSize":"Standard_D2_v2",
```
Comment (bmoore-msft) [inline ansible-tower-rhel/azuredeploy.json]: check for single use vars

File: ansible-tower-rhel/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,728 @@
...
+      "clientStorageAccountType":"Standard_GRS",
+      "diagStorageAccountName":"[concat('dgstorage',uniqueString(resourceGroup().id))]",
+      "ansibleVnetName":"ansible-vnet",
+      "ansibleVnetPrefix":"10.0.0.0/16",
+      "ansibleVnetTowerSubnetName":"tower-subnet",
+      "ansibleVnetTowerSubnetPrefix":"10.0.0.0/24",
+      "ansibleVnetClientSubnetName":"client-subnet",
+      "ansibleVnetClientSubnetPrefix":"10.0.1.0/24",
+      "ansibleVnetId":"[resourceId('Microsoft.Network/virtualNetworks', variables('ansibleVnetName'))]",
+      "towerSubnetRef":"[concat(variables('ansibleVnetId'), '/subnets/', variables('ansibleVnetTowerSubnetName'))]",
```
Comment (bmoore-msft) [inline ansible-tower-rhel/azuredeploy.json]: simplify, just use resourceId

File: ansible-tower-rhel/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,728 @@
...
+                     "publicIPAddress":{
+                        "id":"[resourceId('Microsoft.Network/publicIPAddresses', variables('pipClient02Name'))]"
+                     }
+                  }
+               }
+            ]
+         }
+      },
+      {
+         "type":"Microsoft.Compute/virtualMachines",
```
Comment (bmoore-msft) [inline ansible-tower-rhel/azuredeploy.json]: do client VMs need in-guest config?

Also, consider a copy loop for the 2 client VMs - removes a lot of code...

File: ansible-tower-rhel/README.md (hunk 5 above)
Comment (amalik99) [inline ansible-tower-rhel/README.md]: No, This is fixed by Ansible.

File: ansible-tower-rhel/azuredeploy.json (hunk 8 above)
Comment (amalik99) [inline ansible-tower-rhel/azuredeploy.json]: Client VMs in-guest config is being done using custom script by AnsibleTower. 

Configured copy loop for client VMs, Count is configured as hard-coded value as this is fixed for this particular environment.

**Comment by amalik99** [comment]
Hi Brian, Thanks for review. All requested changes have been implemented.

Let me know in case of any other suggestions.

**Pull request merged**